lib/index.js

```diff
@@ -66,27 +66,26 @@ const internals = {
 };
 
 
-exports.Boom = class extends Error {
+exports.Boom = class Boom extends Error {
 
-    constructor(messageOrError, options = {}) {
+    data = null;
+    output;
 
-        if (messageOrError instanceof Error) {
-            return exports.boomify(Hoek.clone(messageOrError), options);
-        }
+    get isServer() {
 
-        const { statusCode = 500, data = null, ctor = exports.Boom } = options;
-        const error = new Error(messageOrError ? messageOrError : undefined);         // Avoids settings null message
-        Error.captureStackTrace(error, ctor);                           // Filter the stack to our external API
-        error.data = data;
-        const boom = internals.initialize(error, statusCode);
+        return this.output.statusCode >= 500;
+    }
 
-        Object.defineProperty(boom, 'typeof', { value: ctor });
+    set isServer(value) {}                                             // Allow for compatiblity with legacy boom
 
-        if (options.decorate) {
-            Object.assign(boom, options.decorate);
-        }
+    constructor(message, options = {}) {
+
+        const { statusCode = 500, data, headers, ctor = exports.Boom } = options;
+
+        super(message ?? internals.codes.get(statusCode) ?? 'Unknown', options);
```

Comment:
In order to avoid potential conflicts with future values of the second constructor argument, shouldn't we only set options known to be valid (only `cause` at the moment)?

```suggestion
        const { statusCode = 500, data, headers, ctor = exports.Boom, cause } = options;

        super(message ?? internals.codes.get(statusCode) ?? 'Unknown', { cause });
```

or at least remove "our" options?

```suggestion
        const { statusCode = 500, data, headers, ctor = exports.Boom, ...errOpts } = options;

        super(message ?? internals.codes.get(statusCode) ?? 'Unknown', errOpts);
```

The latter option allows to be future proof in terms of code & typings:

```ts

type Options<Data> = {
  // "cause" is *not* in here
}

type ErrorOptions = ConstructorParameters<typeof Error>[1]

export class Boom<Data = any> extends Error {
    constructor(message?: string | Error, options?: Options<Data> & ErrorOptions);
}

```